src/tests/BaGet.Tests/TestServerBuilder.cs

```diff
@@ -24,47 +26,70 @@ public class TestServerBuilder
+        /// Not configured Test Server Builder
         /// </summary>
         /// <returns></returns>
-        public static TestServerBuilder Empty()
+        public static TestServerBuilder Create()
         {
             return new TestServerBuilder();
         }
 
-        public TestServer Build()
+        private readonly string DatabaseTypeKey = $"{nameof(BaGetOptions.Database)}:{nameof(DatabaseOptions.Type)}";
+        private readonly string ConnectionStringKey = $"{nameof(BaGetOptions.Database)}:{nameof(DatabaseOptions.ConnectionString)}";
+        private readonly string StorageTypeKey = $"{nameof(BaGetOptions.Storage)}:{nameof(StorageOptions.Type)}";
+        private readonly string MirrorEnableReadThroughCachingKey = $"{nameof(BaGetOptions.Mirror)}:{nameof(MirrorOptions.EnableReadThroughCaching)}";
+        private readonly string FileSystemStoragePathKey = $"{nameof(BaGetOptions.Storage)}:{nameof(FileSystemStorageOptions.Path)}";
+
+        /// <summary>
+        /// Creates a subdirectory: Path.GetTempPath() + Guid  
+        /// and uses this as location for 
+        /// - Sqlite file  => .\BaGet.db 
+        /// - FilePackageStorageService => .\Packages\*.*
+        /// </summary>
+        /// <returns></returns>
+        public TestServerBuilder UseEmptyTempFolder()
         {
-            
-            Dictionary<string, string> testHostConfig = new Dictionary<string, string>();
-            testHostConfig.Add("Database:Type", "Sqlite");
-            string resolvedSqliteFile = Path.GetFullPath("..\\..\\baget.db");
+            Configuration.Add(DatabaseTypeKey, DatabaseType.Sqlite.ToString());
+            string uniqueTempFolder = Path.Combine(Path.GetTempPath(), System.Guid.NewGuid().ToString("N"));
+            Directory.CreateDirectory(uniqueTempFolder);
+            string resolvedSqliteFile = Path.Combine(uniqueTempFolder, "BaGet.db");
+            string storageFolderPath = Path.Combine(uniqueTempFolder, FilePackageStorageService.DefaultPackagesFolderName);
 
-            testHostConfig.Add("Database:ConnectionString", string.Format("Data Source={0}", resolvedSqliteFile));
-            testHostConfig.Add("Storage:Type", "FileSystem");
-            testHostConfig.Add("Mirror:EnableReadThroughCaching", false.ToString());
+            Configuration.Add(ConnectionStringKey, string.Format("Data Source={0}", resolvedSqliteFile));
+            Configuration.Add(StorageTypeKey, StorageType.FileSystem.ToString());
+            Configuration.Add(MirrorEnableReadThroughCachingKey, false.ToString());
+            Configuration.Add(FileSystemStoragePathKey, storageFolderPath);
+            return this;
+        }
 
-            IConfigurationBuilder configurationBuilder = new ConfigurationBuilder().AddInMemoryCollection(testHostConfig);
+        public TestServer Build()
+        {
+            IConfigurationBuilder configurationBuilder = new ConfigurationBuilder().AddInMemoryCollection(this.Configuration);
```

Comment:
>this [](start = 106, length = 4)

`this.Configuration` -> `Configuration`

Reply:
changed - but personally i don't like that rule!

"this." gives me a lot of context (instance, or static/singleton) on reading/debugging the code!

Reply:
That’s true. It’s a trade off between clarity and verbosity.

That being said, these are the .NET team’s style rules. BaGet is a playground for ideas that may be merged back into the NuGet project, which follows the .NET style guide. Thus, I’d like to stick to their style.

Reply:
That’s true. It’s a trade off between clarity and verbosity.

That being said, these are the .NET team’s style rules. BaGet is a playground for ideas that may be merged back into the NuGet project, which follows the .NET style guide. Thus, I’d like to stick to their style.